### PR TITLE
added comments, updated readme for ene_008 script

### PR DESCRIPTION
**Pull request by Taufiq06**

### CODE REVIEW

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 1]
```diff
@@ -36,116 +40,182 @@
...
                                     'coordinates': [float(lon), float(lat)]
                                 }
+                                # add geojson geometry to the list of data from this row
                                 new_row.append(geometry)
                             else:
+                                # log if no lat lon available and store None for geometry
                                 logging.debug('No lat long available for this data point - skipping!')
                                 new_row.append(None)
                         else:
                             # To fix trouble w/ cartosql not being able to handle '' for numeric:
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:206]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 2]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                                 loc=idx[field]-offset_factor
                                 if loc>0:
+                                    # insert value for columns to the new_row list 
+                                    # if the column is empty, insert None
                                     val = row[loc] if row[loc] != '' else None
                                     new_row.append(val)
                             except IndexError:
                                 pass
+                        # check if last row in new_rows matches with the new_row list
                         new_rows[-1]==new_row
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:250]: I don't think this is doing anything right?
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 3]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                         '''
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:251]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 4]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                         for item in row[1:]:
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:252]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 5]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                             val = row[idx[field]] if row[idx[field]] != '' else None
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:253]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 6]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                             new_row.append(val)
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:254]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 7]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                             new_rows[-1].append(item)
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:255]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py [hunk 8]
```diff
@@ -155,117 +225,140 @@ def processData(existing_ids):
...
                         '''
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:256]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/README.md [hunk 9]
```diff
@@ -1,9 +1,13 @@
 ## Oil and Chemical Spill Incidents Dataset Near Real-time Script
 This file describes the near real-time script that retrieves and processes the [Raw Incident Data dataset](https://incidentnews.noaa.gov/) for [display on Resource Watch](https://resourcewatch.org/data/explore/US-Oil-and-Chemical-Spills).
 
-{Describe how the original data came from the source.}
+This dataset was provided by the source as a CSV file. This CSV was transformed into a table so that it could be uploaded to Carto. In order to transform the data from the CSV to the data table used by Resource Watch, the following changes were made:
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/README.md:4]:
```suggestion
This dataset was provided by the source as a CSV file. In order to transform the data from the CSV to the data table used by Resource Watch, the following changes were made:
```

File: ene_008_oil_chemical_spill_incidents/README.md [hunk 10]
```diff
@@ -1,9 +1,13 @@
...
+- The latitude and longitude from the columns 'lat', 'lon' were used to create the geometry shown on Resource Watch.
+- Datetime was obtained from the column 'open_date'.
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/README.md:6]:
```suggestion
```

File: ene_008_oil_chemical_spill_incidents/README.md [hunk 11]
```diff
@@ -1,9 +1,13 @@
...
+- A unique ID for each event was created based on the 'id' column of the CSV. This was stored in a new column called 'uid'.
+- The source CSV has some entries with breaks in the last column, which are interpreted as an individual row while processing through csv Python library. Before processing each row, it was checked if the unique id in the row could be converted to an integer to make sure it was a new row.
```
Comment (amsnyder) [inline ene_008_oil_chemical_spill_incidents/README.md:8]:
```suggestion
- The source CSV has some entries with breaks in the last column, which are interpreted as a new row while processing through csv Python library. These breaks were identified and removed.
```

File: ene_008_oil_chemical_spill_incidents/contents/src/__init__.py (hunk 2 above)
Comment (Taufiq06) [reply ene_008_oil_chemical_spill_incidents/contents/src/__init__.py:250]: right. the author was probably debugging it. so, we can get rid of it.

**Pull request merged**